components/Ink/src/MDCInkView.m

```diff
@@ -193,9 +205,27 @@ - (CGFloat)maxRippleRadius {
 }
 
 - (void)setMaxRippleRadius:(CGFloat)radius {
-  if (self.inkLayer.maxRippleRadius != radius) {
+  // Keep track of the set value in case the caller will change inkStyle later
+  _maxRippleRadius = radius;
+  if (MDCCGFloatEqual(self.inkLayer.maxRippleRadius, radius)) {
+    return;
+  }
+
+  // Legacy Ink updates inkLayer.maxRippleRadius regardless of inkStyle
+  if (self.usesLegacyInkRipple) {
     self.inkLayer.maxRippleRadius = radius;
     [self setNeedsLayout];
+  } else {
+    // New Ink Bounded style ignores maxRippleRadius
+    switch(self.inkStyle) {
+      case MDCInkStyleUnbounded:
+        self.inkLayer.maxRippleRadius = radius;
+        [self setNeedsLayout];
```

Comment:
Why does this require a setNeedsLayout?
If we need one here, do we need one where we set self.inkLayer.maxRippleRadius above?

Reply:
Done. It's needed for legacy Ink but not new Ink.